maintainer/envs/minimal_requirements.txt

```diff
@@ -0,0 +1,18 @@
+biopython
+codecov
+cython
+duecredit
```

Comment:
I'm pretty sure you're changing the Azure run here--for example, I was not using `duecredit` on purpose to avoid the annoying spam at the end of the test run.

I don't have specific views on the `conda` installs, but for `pip` I'm a bit more concerned here.

More broadly, I'm also a little concerned that the abstractions may not really be worth it unless they are for standardized things like [PEP 518](https://www.python.org/dev/peps/pep-0518/), which is indeed precisely for "Specifying Minimum Build System Requirements for Python Projects." For that, `pyproject.toml` should be used.

Reply:
There's also "bare minimum to build," "bare minimum to test," and what if I want to use a slightly different dependency pin for a while on 32-bit Windows because of some issue? The abstraction to a file becomes a burden then if another CI service/run is using it.

The abstraction vs. convenience to just edit in the CI yml itself is something that should probably at least be discussed.

Not saying we shouldn't do this, but I do have my reservations about some aspects.